src/Message/Headers.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace Jasny\HttpMessage\Message;
+
+use Jasny\HttpMessage\Headers as HeaderObject;
+
+/**
+ * ServerRequest header methods
+ */
+trait Headers
+{
+    /**
+     * HTTP headers
+     *
+     * @var object Headers Class
+     */
+    protected $headers;
+
+    /**
+     * Public function to create header object 
+     * 
+     */
+    public function initHeaders()
+    {
+        $this->headers = new HeaderObject($this->determineHeaders());
```

Comment:
Preferably do an `if (isset($this->headers))` and call `initHeaders()` when using the headers (in `getHeaders()`, `addHeader()`, etc) rather than in the constructor.